File: SmartERP/SmartERP.Web/Modules/Financial/FinancialPermissionKeys.cs

using Serenity.ComponentModel;
using System.ComponentModel;

namespace SmartERP.Financial
{
    [NestedPermissionKeys]
    [DisplayName("Financial")]
    public class PermissionKeys
    {

        [DisplayName("Chart Of Accounts")]
        public class Accounts
        {
            public const string View = "Financial:Chart Of Accounts:View";
            public const string Create = "Financial:Chart Of Accounts:Create";
            public const string Update = "Financial:Chart Of Accounts:Update";
            public const string Delete = "Financial:Chart Of Accounts:Delete";
        }
        [Description("Opening Balance")]
        public const string OpeningBalance = "Financial:Opening Balance";
        [Description("Debit Voucher")]
        public const string DebitVoucher = "Financial:Debit Voucher";
        [Description("Credit Voucher")]
        public const string CreditVoucher = "Financial:Credit Voucher";
        [Description("Contra Voucher")]
        public const string ContraVoucher = "Financial:Contra Voucher";
        [Description("Journal Voucher")]
        public const string JournalVoucher = "Financial:Journal Voucher";

        [Description("Transactions")]
        public const string Transactions = "Financial:Transactions";
    }
}

File: SmartERP/SmartERP.Web/Imports/ServerTypings/Financial.PermissionKeys.ts
namespace SmartERP.Financial {
    declare namespace PermissionKeys {
        export const OpeningBalance = "Financial:Opening Balance";
        export const DebitVoucher = "Financial:Debit Voucher";
        export const CreditVoucher = "Financial:Credit Voucher";
        export const ContraVoucher = "Financial:Contra Voucher";
        export const JournalVoucher = "Financial:Journal Voucher";
        export const Transactions = "Financial:Transactions";

        namespace Accounts {
            export const View = "Financial:Chart Of Accounts:View";
            export const Create = "Financial:Chart Of Accounts:Create";
            export const Update = "Financial:Chart Of Accounts:Update";
            export const Delete = "Financial:Chart Of Accounts:Delete";
        }
    }
}
